src/library/returnBook/ReturnBookControl.java

```diff
@@ -0,0 +1,95 @@
+package library.returnBook;
+import library.entities.Item;
+import library.entities.Library;
+import library.entities.Loan;
+
+public class ReturnBookControl {
+
+    private ReturnBookUI ui;
+
+    private enum ControlState { INITIALISED, READY, INSPECTING };
+
+    private ControlState state;
+    
+    private Library library;
+
+    private Loan currentLoan;
+    
+
+    public ReturnBookControl() {
+        this.library = Library.getInstance();
+        state = ControlState.INITIALISED;
+    }
+    
+    
+    public void setUI(ReturnBookUI ui) {
+        if (!state.equals(ControlState.INITIALISED)) {
+            throw new RuntimeException("ReturnBookControl: cannot call setUI except in INITIALISED state");
+        }
+
+        this.ui = ui;
+        ui.SeTrEaDy();
+        state = ControlState.READY;
+    }
+
+
+    public void bookScanned(long bookId) {
```

Comment:
#PR41 - D4: 
Please rename to itemScanned as in the BRLS class diagram.

Reply:
https://github.com/elliotmitchum/ITC205/pull/41/commits/94485e68568dcdfbafb265738c2ed2b3f645ce28